lib/dpl/provider/pages.rb

```diff
@@ -0,0 +1,70 @@
+module DPL
+  class Provider
+    class Pages < Provider
+      """Implements Github Pages deployment
+
+      Options:
+        - repo [optional, for pushed to other repos]
+        - github-token [required]
+        - target-branch [optional, defaults to gh-pages]
+        - local-dir [optional, defaults to `pwd`]
+        - fqdn [optional]
+        - project-name [optional, defaults to fqdn or repo slug]
+        - email [optional]
+        - name [optional]
+      """
+
+      def initialize(context, options)
+        super
+
+        @build_dir = options[:local_dir] || '.'
+
+        @project_name = options[:project_name] || fqdn || slug
+        @gh_fqdn = fqdn
+
+        @gh_ref = "github.com/#{slug}.git"
+        @target_branch = options[:target_branch] || 'gh-pages'
+        @gh_token = option(:github_token)
+
+        @gh_email = options[:email] || 'deploy@travis-ci.org'
+        @gh_name = "#{options[:name] || 'Deployment Bot'} (from Travis CI)"
+      end
+
+      def fqdn
+        options.fetch(:fqdn) { nil }
+      end
+
+      def slug
+        options.fetch(:repo) { context.env['TRAVIS_REPO_SLUG'] }
+      end
+
+      def check_auth
+      end
+
+      def needs_key?
+        false
+      end
+
+      def github_deploy
+        context.shell 'rm -rf .git > /dev/null 2>&1'
```

Comment:
Good point! I'll do my best to avoid this :)
I think, I'll copy build folder to somewhere under `/tmp`. Do you think this would work?